include/selene/State.h

```diff
@@ -134,9 +134,14 @@ class State {
     }
 
     bool operator()(const char *code) {
-        bool result = !luaL_dostring(_l, code);
-        if(result) lua_settop(_l, 0);
-        return result;
+        ResetStackOnScopeExit savedStack(_l);
+        int status = luaL_dostring(_l, code);
+        if(status) {
+            _exception_handler->Handle_top_of_stack(status, _l);
+            return false;
+        }
+        savedStack.KeepChanges();
```

Comment:
The member functions `Load(...)`, `OpenLib(...)`, `Push(...)` and `operator()(...)` leave objects on the stack. There is no function to remove objects from the stack.
 1. Make them to not leave objects on the stack and remove `Push(...)` and `Read(...)`.
 2. Add `Pop(...)` or similar functions.
I see no real use of managing objects on the stack at this level of abstraction - apart from using Lua's implicit type conversions. The latter seems unimportant enough to me to favor item 1.
@jeremyong: Am I missing a use case?